site/content/docs/developing/overrides/cdk.md

```diff
@@ -0,0 +1,176 @@
+# CDK Overrides
+
+{% include 'overrides-intro.md' %}
+
+## When should I use CDK overrides over YAML patch?
+
+Both options are a "break the glass" mechanism to access and configure functionality that is not surfaced by Copilot [manifests](../../manifest/overview.en.md).
+
+We recommend using the AWS Cloud Development Kit (CDK) overrides over [YAML patches](./yamlpatch.md) if you'd like to leverage
+the expressive power of a programming language. The CDK allows you to add safe and powerful modifications to your CloudFormation template.
+
+## How to get started
+
+You can extend your CloudFormation template with the CDK by running the `copilot [noun] override` command.
+For example, you can run `copilot svc override` to update the template of a Load Balanced Web Service.
+
+The command will generate a new CDK application under the `copilot/[name]/override` directory with the following structure:
+```console
+.
+├── bin/
+│   └── override.ts
+├── .gitignore
+├── cdk.json
+├── package.json
+├── README.md
+├── stack.ts
+└── tsconfig.json
+```
+
+You can get started by editing the `stack.ts` file. For example, if you decided to override the ECS service properties
+with `copilot svc override`, the following `stack.ts` file will be generated for you to modify:
+
+```typescript
+import * as cdk from 'aws-cdk-lib';
+import { aws_ecs as ecs } from 'aws-cdk-lib';
+
+export class TransformedStack extends cdk.Stack {
+    constructor (scope: cdk.App, id: string, props?: cdk.StackProps) {
+        super(scope, id, props);
+         this.template = new cdk.cloudformation_include.CfnInclude(this, 'Template', {
+            templateFile: path.join('.build', 'in.yaml'),
+        });
+        this.appName = template.getParameter('AppName').valueAsString;
+        this.envName = template.getParameter('EnvName').valueAsString;
+
+        this.transformService();
+    }
+ 
+    // TODO: implement me.
+    transformService() {
+      const service = this.template.getResource("Service") as ecs.CfnService;
+      throw new error("not implemented");
+    }
+}
+```
+
+## How does it work?
+
+As can be seen in the above `stack.ts` file, Copilot will use the [clouformation_include module](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.cloudformation_include-readme.html) 
+provided by the CDK to help author transformations. This library is the CDK’s [recommendation](https://docs.aws.amazon.com/cdk/v2/guide/use_cfn_template.html) from their 
+"Import or migrate an existing AWS CloudFormation template" guide. It enables accessing the resources not surfaced by the Copilot manifest as 
+[L1 constructs](https://docs.aws.amazon.com/cdk/v2/guide/constructs.html) and ensures that the CDK retains the resources on synthesis.  
```

Comment:
> ensures that the CDK retains the resources on synthesis

what does this mean 🥺

Reply:
Removed! Didn't mean to scare anyone haha, it was more internal knowledge. It didn't mean anything important, just that everything defined in the CFN template would be mapped properly as an L1 CDK construct